src/test/java/org/eclipse/uprotocol/communication/InMemorySubscriberTest.java

```diff
@@ -31,144 +50,664 @@
 import org.eclipse.uprotocol.v1.UStatus;
 import org.eclipse.uprotocol.v1.UUri;
 
-
+@ExtendWith(MockitoExtension.class)
 public class InMemorySubscriberTest {
+
+    @Mock
+    private UTransport transport;
+
+    @Mock
+    private InMemoryRpcClient rpcClient;
+
+    @Mock
+    private SimpleNotifier notifier;
+
+    private final UUri topic = UUri.newBuilder().setAuthorityName("hartley").setUeId(3)
+        .setUeVersionMajor(1).setResourceId(0x8000).build();
+
+    private final UUri source = UUri.newBuilder().setAuthorityName("Hartley").setUeId(4)
+        .setUeVersionMajor(1).build();
+    private final UListener listener = new UListener() {
+        @Override
+        public void onReceive(UMessage message) {
+            // Do nothing
+        }
+    };
+
+
+    @BeforeEach
+    public void setup() {
+        rpcClient = mock(InMemoryRpcClient.class);
+        notifier = mock(SimpleNotifier.class);
+        transport = mock(UTransport.class);
+    }
+
+
     @Test
-    @DisplayName("Test subscribe happy path")
-    public void test_subscribe_happy_path() {
-        UUri topic = createTopic();
-        UTransport transport = new TestUTransport();
-        Subscriber subscriber = new InMemorySubscriber(transport, new InMemoryRpcClient(transport));
-        CompletionStage<SubscriptionResponse> response = subscriber.subscribe(topic, new UListener() {
-            @Override
-            public void onReceive(UMessage message) {
-                // Do nothing
-            }
-        }, null);
-        response.toCompletableFuture().join();
-        assertFalse(response.toCompletableFuture().isCompletedExceptionally());
+    @DisplayName("Testing simple mock of RpcClient and notifier happy path")
+    public void test_simple_mock_of_rpcClient_and_notifier() {
+        
+        final SubscriptionResponse response = SubscriptionResponse.newBuilder()
+        .setTopic(topic)
+        .setStatus(SubscriptionStatus.newBuilder().setState(SubscriptionStatus.State.SUBSCRIBED).build())
+        .build();
+
+        when(transport.registerListener(any(UUri.class), any(UListener.class)))
+            .thenReturn(CompletableFuture.completedFuture(UStatus.newBuilder().setCode(UCode.OK).build()));
+        
+        when(transport.getSource()).thenReturn(source);
+
+        when(rpcClient.invokeMethod(any(UUri.class), any(UPayload.class), any(CallOptions.class)))
+            .thenReturn(CompletableFuture.completedFuture(UPayload.pack(response)));
+
+        when(notifier.registerNotificationListener(any(UUri.class), any(UListener.class)))
+            .thenReturn(CompletableFuture.completedFuture(UStatus.newBuilder().setCode(UCode.OK).build()));
+
+
+        InMemorySubscriber subscriber = new InMemorySubscriber(transport, rpcClient, notifier);
+        assertNotNull(subscriber);
+        
+        assertDoesNotThrow(() -> {
+            assertEquals(subscriber.subscribe(topic, listener).toCompletableFuture().get().getStatus().getState(),
```

Comment:
Swap these 2 arguments so they are in the correct order: expected value, actual value.

Reply:
It appears that the entire up-java has them swapped. I kept that from what Tamara originally did but didn't fix it, I'll have to do a major clean later (not now)